src/canvas/PatternPaths/PatternPath.tsx

```diff
@@ -42,143 +33,40 @@ export abstract class PatternPath implements IPatternPath {
-        const myLastPoint = this._points[this._points.length - 1];
-        // Radius to check within to see if we should snap to point.
-        const radius = 10;
-        let updatedFirstPoint = false;
-        let updatedLastPoint = false;
-
-        for (let i = 0; i < paths.length; i++) {
-            const path = paths[i];
-            if (path === this) {
-                continue;
-            }
-            const points = path.getPoints();
-
-            const otherFirstPoint = points[0];
-            const otherLastPoint = points[points.length - 1];
-
-            if(!updatedFirstPoint && myFirstPoint.isWithinRadius(otherFirstPoint, radius)) {
-                this.snapStartToPoint(otherFirstPoint);
-                updatedFirstPoint = true;
-            }
-
-            if(!updatedFirstPoint && myFirstPoint.isWithinRadius(otherLastPoint, radius)) {
-                this.snapStartToPoint(otherLastPoint);
-                updatedFirstPoint = true;
-            }
-
-            if(!updatedLastPoint && myLastPoint.isWithinRadius(otherFirstPoint, radius)) {
-                this.snapEndToPoint(otherFirstPoint);
-                updatedLastPoint = true;
-            }
-
-            if(!updatedLastPoint && myLastPoint.isWithinRadius(otherLastPoint, radius)) {
-                this.snapEndToPoint(otherLastPoint);
-                updatedLastPoint = true;
-            }
-
-        }
-
-        if (updatedFirstPoint || updatedLastPoint) {
-            this._updatePath2D();
-        }
-    };
-
     getLengthInPixels = (): number => {
-        if (!this._fittedSegment) {
-            throw new Error();
-        }
-        return this._fittedSegment.getLength();
+        return this._segment.getLength();
     };
 
     getSegment = (): Segment => {
-        if (!this._fittedSegment) {
-            throw new Error();
-        }
-        return this._fittedSegment;
+        return this._segment;
     };
 
-    setFittedSegment = (): void => {
-        this._setFittedSegment();
-        this._points = this._fittedSegment ? this._fittedSegment.computePoints() : new Array<Point>();
-    };
-
-    splitAtPoint = (interesection: Point): PatternPath[] => {
+    splitAtPoint = (intersection: Point): PatternPath[] => {
         const originalSegment = this.getSegment();
-        const splitSegments = originalSegment.split(interesection);
+        const splitSegments = originalSegment.split(intersection);
         if (splitSegments.length !== 2) {
             throw new Error("Split did not return correct number of segments");
         }
 
         return this._createPathsFromSegments(splitSegments);
     };
 
-    protected abstract _createPathsFromSegments(segments: Segment[]): PatternPath[];
-    protected abstract _setFittedSegment(): void;
-    protected abstract _updatePath2D(): void;
-    protected abstract _addPoint(point: Point): void;
+    protected _createPathsFromSegments = (segments: Segment[]): PatternPath[] => {
```

Comment:
Maybe update name to _createPatterPathsFromSegments

Reply:
yes, will do.